server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorEncoderDecoder.java

```diff
@@ -70,7 +67,7 @@ public static void decodeDenseVector(BytesRef vectorBR, float[] vector) {
         }
         ByteBuffer byteBuffer = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length);
         for (int dim = 0; dim < vector.length; dim++) {
-            vector[dim] = byteBuffer.getFloat();
+            vector[dim] = byteBuffer.getFloat((dim + vectorBR.offset) * Float.BYTES);
```

Comment:
I would expect that `vectorBR.offset` is not needed here? In fact it would be a problem if it is anything other than `0`?

Reply:
@ChrisHegarty I guess I misunderstand. I thought `getFloat` was the absolute position within the underlying `bytes`, not relative to the starting position (dictated by the wrapping of bytes with offset).

Reply:
My equation is wrong for sure, I think it should be `(dim * Float.BYTES) + vectorBR.offset`?

Reply:
This test passes with my fixed equation:

```
 public void testDecoding() {
        float[] inputFloats = new float[]{1f, 2f, 3f, 4f};
        ByteBuffer byteBuffer = ByteBuffer.allocate(16);
        for (float f : inputFloats) {
            byteBuffer.putFloat(f);
        }
        BytesRef floatBytes = new BytesRef(byteBuffer.array());
        floatBytes.length = 12;
        floatBytes.offset = 4;
        float[] outputFloats = new float[3];
        VectorEncoderDecoder.decodeDenseVector(floatBytes, outputFloats);
        assertArrayEquals(outputFloats, new float[]{2f, 3f, 4f}, 0f);
    }
```

Reply:
@ChrisHegarty what do you think? I fixed my math. Is this still an issue?